src/main/kotlin/service/UserService.kt

```diff
@@ -60,4 +64,24 @@ class UserService(
         users.forEach { builder.addUsers(it.toGrpcUser()) }
         return builder.build()
     }
+
+    /**
+     * Service implementation of [SnowballRService.softDeleteUser].
+     */
+    override suspend fun softDeleteUser(request: Base.Id): Base.Nothing {
+        val currentUser = repo.getUserById(dummyUserId!!)
+        val userToDelete = repo.getUserById(request.id)
+
```

Comment:
`currentUser.id.toString() != request.id` is used twice. We could write something like this:
```suggestion
val isSameUser = currentUser.id == userToDelete.id
```
and then use `isSameUser` in the checks. What are your thoughts on this?

Reply:
Is there a reason why you kept the string comparison instead of comparing the UUIDs directly?

Reply:
No, I just had a fault in the delete user test, so it didn't work without calling the `toString()`. Now it works.